### PR TITLE
Extract localization utilities to a new project.

### PR DESCRIPTION
**Pull request by iclanton**

### CODE REVIEW

**Review by D4N14L — APPROVED** [review]

File: libraries/localization-utilities/src/LocFileParser.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    parsedFile = JsonFile.parseString(options.content);
+    try {
+      LOC_JSON_SCHEMA.validateObject(parsedFile, options.filePath);
+    } catch (e) {
+      options.terminal.writeError(`The loc file is invalid. Error: ${e}`);
+    }
+  } else if (/\.resjson$/i.test(options.filePath)) {
+    const resjsonFile: Record<string, string> = JsonFile.parseString(options.content);
+    parsedFile = {};
+    const comments: Record<string, string> = {};
```
Comment (dmichon-msft) [inline libraries/localization-utilities/src/LocFileParser.ts:59]:
```suggestion
    const comments: Map<string, string> = new Map();
```

File: libraries/localization-utilities/src/LocFileParser.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    for (const [key, value] of Object.entries(resjsonFile)) {
+      if (key.startsWith('_') && key.endsWith('.comment')) {
+        const commentKey: string = key.substring(1, key.length - '.comment'.length);
+        comments[commentKey] = value;
```
Comment (dmichon-msft) [inline libraries/localization-utilities/src/LocFileParser.ts:63]:
```suggestion
        comments.set(commentKey, value);
```

File: libraries/localization-utilities/src/LocFileParser.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      } else {
+        parsedFile[key] = { value };
+      }
+    }
+
+    const orphanComments: string[] = [];
+    for (const [key, comment] of Object.entries(comments)) {
```
Comment (dmichon-msft) [inline libraries/localization-utilities/src/LocFileParser.ts:70]:
```suggestion
    for (const [key, comment] of comments) {
```

**Review by dmichon-msft — APPROVED** [review]
Few minor notes

File: libraries/localization-utilities/package.json [hunk 4]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "@rushstack/localization-utilities",
+  "version": "0.0.0",
```
[the remark this answers is not in the document]
Comment (iclanton) [reply libraries/localization-utilities/package.json:3]: Yeah. Publish will bump this to `0.1.0`.

File: libraries/localization-utilities/package.json [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    "directory": "libraries/localization-utilities"
+  },
+  "scripts": {
+    "build": "heft build --clean",
+    "_phase:build": "heft build --clean"
+  },
+  "dependencies": {
+    "@rushstack/node-core-library": "workspace:*",
+    "@rushstack/typings-generator": "workspace:*",
+    "@types/node": "12.20.24",
```
[the remark this answers is not in the document]
Comment (iclanton) [reply libraries/localization-utilities/package.json:20]: Not actually needed by the public API.

**Pull request merged**